src/components/ProductVersionCreateEditPage/ProductVersionCreateEditPage.tsx

```diff
@@ -0,0 +1,199 @@
+
+import * as productVersionApi from 'services/productVersionApi';
+
+import { validateProductVersionName } from 'utils/formValidationHelpers';
+import { createSafePatch } from 'utils/patchHelper';
+import { generatePageTitle } from 'utils/titleHelper';
+
+const formConfig = {
+  version: {
+    isRequired: true,
+
+    validators: [
+      {
+        validator: validateProductVersionName,
+        errorMessage: 'The version should consist of two numeric parts separated by a dot (e.g. 1.0).',
+      },
+    ],
+  },
+  'attributes.brewTagPrefix': {
+    isRequired: true,
+  },
+};
+
+interface IProductVersionCreateEditPageProps {
+  isEditPage?: boolean;
+}
+
+export const ProductVersionCreateEditPage = ({ isEditPage = false }: IProductVersionCreateEditPageProps) => {
+  const { productId } = useParams();
+  const { productVersionId } = useParams();
+  const navigate = useNavigate();
+
+  // create page
+  const serviceContainerCreatePage = useServiceContainer(productVersionApi.createProductVersion);
+
+  // edit page - get method
+  const serviceContainerEditPageGet = useServiceContainer(productVersionApi.getProductVersion);
+  const serviceContainerEditPageGetRunner = serviceContainerEditPageGet.run;
+
+  // edit page - patch method
+  const serviceContainerEditPagePatch = useServiceContainer(productVersionApi.patchProductVersion);
+
+  const { register, setFieldValues, getFieldState, getFieldErrors, handleSubmit, isSubmitDisabled } = useForm();
+
+  useTitle(
+    generatePageTitle({
+      pageType: isEditPage ? 'Edit' : 'Create',
+      serviceContainer: serviceContainerEditPageGet,
+      firstLevelEntity: 'Product',
+      nestedEntity: 'Version',
+      entityName: `${serviceContainerEditPageGet.data?.version} ${PageTitles.delimiterSymbol} ${serviceContainerEditPageGet.data?.product.name}`,
+    })
+  );
+
+  const submitCreate = (data: IFieldValues) => {
+    return serviceContainerCreatePage
+      .run({
+        serviceData: { data: { ...data, product: { id: productId } } },
+      })
+      .then((response: any) => {
+        const newProductVersionId = response?.data?.id;
+        if (!newProductVersionId) {
+          throw new Error(`Invalid newProductVersionId coming from Orch POST response: ${newProductVersionId}`);
+        }
+        navigate(`../${newProductVersionId}`);
+      })
+      .catch((e: any) => {
+        throw new Error('Failed to create Product Version.');
+      });
+  };
+
+  const submitUpdate = (data: IFieldValues) => {
+    const patchData = createSafePatch(serviceContainerEditPageGet.data, {
+      version: data.version,
+      attributes: { BREW_TAG_PREFIX: data['attributes.brewTagPrefix'] },
+    });
+
+    serviceContainerEditPagePatch
+      .run({ serviceData: { id: productVersionId, patchData } })
+      .then(() => {
+        navigate(`..`);
+      })
+      .catch(() => {
+        throw new Error('Failed to edit Product Version.');
```

Comment:
same with error above, consider `uiLogger`

Reply:
This issue will be considered in different Jira (NCL-8169).